src/sinks/uds.rs

```diff
@@ -164,29 +196,41 @@ impl MetricSink for BufferedUdsMetricSink {
 #[cfg(test)]
 mod tests {
     use super::{BufferedUdsMetricSink, MetricSink, UdsMetricSink};
-    use std::os::unix::net::UnixStream;
+    use std::os::unix::net::UnixDatagram;
+    use tempdir::TempDir;
 
     #[test]
     fn test_uds_metric_sink() {
-        let (socket, _recv) = UnixStream::pair().unwrap();
-        let sink = UdsMetricSink::from(socket);
+        let socket = UnixDatagram::unbound().unwrap();
+        let tmp_dir = TempDir::new("testing").unwrap();
+        let file_path = tmp_dir.path().join("tmp.sock");
+        // Create a listener.
+        let _listener = UnixDatagram::bind(&file_path);
+        let sink = UdsMetricSink::from(socket, file_path);
         assert_eq!(7, sink.emit("buz:1|m").unwrap());
     }
 
     #[test]
-    fn test_non_blocking_udp_metric_sink() {
-        let (socket, _recv) = UnixStream::pair().unwrap();
-        socket.set_nonblocking(true).unwrap();
-        let sink = UdsMetricSink::from(socket);
+    fn test_non_blocking_uds_metric_sink() {
```

Comment:
Looks like the call to `.set_nonblocking()` got lost?

Reply:
Added it back.